### PR TITLE
Optimistically detect if params are coming from get_aws_connection_in…

### PR DESCRIPTION
**Pull request by sheepkiller**
…fo()
As discussed in dev mailing lists (https://groups.google.com/forum/#!topic/ansible-devel/cAJTW3V4dSk) current version of boto3_conn seems to overrides **params from get_aws_connection_info() which is already boto3 friendly.
Here's a very simple patch to workaround the problem. I deliberately avoid to make the assumption that get_aws_connection_info() is systematically used.

Thanks

### CODE REVIEW

File: lib/ansible/module_utils/ec2.py [hunk 1]
```diff
@@ -41,9 +41,13 @@
 
 
 def boto3_conn(module, conn_type=None, resource=None, region=None, endpoint=None, **params):
-    profile = params.pop('profile_name', None)
-    params['aws_session_token'] = params.pop('security_token', None)
-    params['verify'] = params.pop('validate_certs', None)
+    # if params are inherited from get_aws_connection_info(), they are already properly set
+    if params['profile_name'] is None:
+        profile = params.pop('profile_name', None)
```
Comment (dblenkus) [inline lib/ansible/module_utils/ec2.py]: Does it make sense to pop `profile_name` from `params` with the default `None` if you already know that `params['profile_name']` is `None`?

Comment (juliedavila) [inline lib/ansible/module_utils/ec2.py]: I'm with @dblenkus it doesn't really make sense at all. If `params['profile_name']` is already `None` then why would you do this over the original line?

**Comment by sheepkiller** [comment]
Thanks for you reviews.
You're right, I completely missed that point. I've updated my branch to incorporate your comments.

**Comment by sheepkiller** [comment]
Hi!
This PR seems stalled due to the lack of a reproducible scenario. I was able to make a playbook to help you to investigate the problem. Unfortunatly, it's too late for 2.0 release. Sorry for the delay.
The environment is archlinux with all packages up-to-date but it was tested at ${WORK} with centos 7 and ubuntu. outputs, policy and (obfuscated) playbook are attached. You also need a trusted user to impersonate route53 role.

[pr-13283.zip](https://github.com/ansible/ansible/files/86444/pr-13283.zip)

Thanks

$ virtualenv2 pr-13283 
$ . pr-13283/bin/activate
(pr-13283)$ cd pr-13283/
(pr-13283)$ pip install -U git+https://github.com/ansible/ansible.git@devel#egg=ansible
[...](pr-13283)$ pip install boto six netaddr boto3 datetime
[...](pr-13283)$ ansible-playbook -i 127.0.0.1 pr-13283.yaml > output-prepatch.txt 2>&1
(pr-13283)$ curl https://patch-diff.githubusercontent.com/raw/ansible/ansible/pull/13283.diff | patch -d lib/python2.7/site-packages -p2
[...](pr-13283)$ ansible-playbook -i 127.0.0.1 pr-13283.yaml > output-postpatch.txt 2>&1
(pr-13283)$ diff -u output-prepatch.txt output-postpatch.txt 
--- output-prepatch.txt 2016-01-11 22:35:07.764510330 +0100
+++ output-postpatch.txt        2016-01-11 22:40:27.966635483 +0100
@@ -15,11 +15,8 @@
 changed: [localhost]

 TASK [Return a count of all hosted zones wit STS] ******************************
-An exception occurred during task execution. To see the full traceback, use -vvv. The error was: botocore.exceptions.ClientError: An error occurred (InvalidClientTokenId) when calling the GetHostedZoneCount operation: The security token included in the request is invalid.

## -fatal: [localhost]: FAILED! => {"changed": false, "failed": true, "parsed": false}

-NO MORE HOSTS LEFT *************************************************************
+ok: [localhost]

 PLAY RECAP *********************************************************************
-localhost                  : ok=4    changed=1    unreachable=0    failed=1  
+localhost                  : ok=5    changed=1    unreachable=0    failed=0

**Comment by sheepkiller** [comment]
Should fix #13727 too

**Comment by defionscode** [comment]
This is good to go in my book

shipit

**Comment by sheepkiller** [comment]
duplicate of #14355

**Comment by juliedavila** [comment]
merge conflicts now

**Comment by sheepkiller** [comment]
note to myself: learn git :)

**Pull request closed without merging**